fpmsyncd/routesync.cpp

```diff
@@ -158,11 +152,199 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
     else
     {
         SWSS_LOG_INFO("Warm-Restart mode: RouteTable set msg: %s %s %s\n",
-                      destipprefix, nexthops.c_str(), ifnames.c_str());
+                      destipprefix, gw_ips.c_str(), if_names.c_str());
 
         const KeyOpFieldsValuesTuple kfv = std::make_tuple(destipprefix,
                                                            SET_COMMAND,
                                                            fvVector);
         m_warmStartHelper.insertRefreshMap(kfv);
     }
 }
+
+/* Handle vnet route */      
+void RouteSync::onVnetRouteMsg(int nlmsg_type, struct nl_object *obj)
+{
+    struct rtnl_route *route_obj = (struct rtnl_route *)obj;  
+
+    /* Only support unicast for VNET routes now. Will update this part in the future */
+    if (rtnl_route_get_type(route_obj) != RTN_UNICAST) {
+        SWSS_LOG_INFO("We only support UNICAST for vnet routes\n");
+        return;
+    }
+
+    /* Get the destination IP prefix */
+    struct nl_addr *dip = rtnl_route_get_dst(route_obj);
+    char destipprefix[MAX_ADDR_SIZE + 1] = {0};
+    nl_addr2str(dip, destipprefix, MAX_ADDR_SIZE);
+
+    /* Get VRF index and VRF name */
+    unsigned int vrf_index = rtnl_route_get_table(route_obj);
+    char vrf_name[IFNAMSIZ] = {0};
+
+    /* If we cannot get the VRF name */
+    if (!getIfName(vrf_index, vrf_name, IFNAMSIZ))
```

Comment:
Not sure if we want to proceed if `vrf_name` is unknown. _Orchagent_ checks for the VRF state and if not found, it just keeps retrying. I would think this is an exceptional case.

Reply:
Yes, my point is, we got a route update on a VRF but the VRF device is not found/deleted. In this case, I would suggest to log error and return.

Reply:
Fixed in https://github.com/Azure/sonic-swss/pull/772/commits/d865382907b9e0f357e2f68594d84b067a23e326